game-app/game-headless/src/main/java/org/triplea/game/server/HeadlessServerStartupRemote.java

```diff
@@ -0,0 +1,129 @@
+package org.triplea.game.server;
+
+import com.google.common.base.Preconditions;
+import games.strategy.engine.framework.HeadlessAutoSaveType;
+import games.strategy.engine.framework.message.PlayerListing;
+import games.strategy.engine.framework.startup.mc.IServerStartupRemote;
+import games.strategy.net.INode;
+import java.io.BufferedInputStream;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.HashSet;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.triplea.io.IoUtils;
+
+@Slf4j
+public class HeadlessServerStartupRemote implements IServerStartupRemote {
+
+  private final ServerModelView serverModelView;
+  private final HeadlessGameServer headlessGameServer;
+
+  public HeadlessServerStartupRemote(
+      ServerModelView serverModelView, HeadlessGameServer headlessGameServer) {
+    this.serverModelView = serverModelView;
```

Comment:
Should this also have Preconditions.checkNotNull()?

Reply:
Thought about it, probably better using it